### PR TITLE
Fix #495: [windows] No browser found

### LINKED ISSUE #495: Regression on Windows: no browser found
acli-missing-browser

### PR DESCRIPTION
**Pull request by danepowell**
**Motivation**
Fixes #495

**Proposed changes**
On Windows, assume `start` exists instead of checking for it (AFAICT, it _always_ exists as a built-in in common shells, and there's not a consistent way of finding commands across shells on Windows).

Also improve error output to guide people towards self-solving the issue.

**Alternatives considered**
Fix `commandExists` to properly find commands on Windows. Good luck with that one, I couldn't untangle how to check that `start` exists, and any solution would probably be shell-specific.

**Testing steps**
1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `acli ckc`
3. Run `auth:login` on Windows and check that it opens a browser window when prompted.

**Merge requirements**
- [x] _Bug_, _enhancement_, or _breaking change_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Comment by anavarre** [comment]
@SophieG this is the PR to try and fix the issue with Windows users not being able to open up links in the default browser. Ideally the customer impacted would try to build from source and check out this PR.

**Review by grasmash — APPROVED** [review]

**Comment by danepowell** [comment]
If we merge https://github.com/acquia/cli/pull/520 and rebase this PR, Sophie could get a pre-built Phar to help with testing

**Comment by SamFoster** [comment]
Guys

may I ask if this fix has made its way into a compiled PHAR file I can get hold of? Or do I need to await the next release - and when will that be?

Obviously I'm asking as I have also got this issue - Windows 10 related

Sam

**Comment by anavarre** [comment]
@SamFoster #520 is exactly what you're asking for. We understand the pain to build from source and the Phar file will make testing way more easier.

**Comment by danepowell** [comment]
@SamFoster @SophieG please download acli.phar from the following link to help test this fix: https://transfer.sh/gXl9K/acli.phar

File: src/Helpers/LocalMachineHelper.php [hunk 1]
```diff
@@ -366,18 +371,20 @@ public function startBrowser($uri = NULL, $browser = NULL): bool {
...
         $browser = 'xdg-open';
       }
       else {
         if ($this->commandExists('open')) {
+          // Darwin.
           $browser = 'open';
         }
         else {
-          if ($this->commandExists('start')) {
+          if (OsInfo::isWindows()) {
```
Comment (anavarre) [inline src/Helpers/LocalMachineHelper.php:383]: Would we be safer if we'd be checking either/or?

**Comment by SamFoster** [comment]
Dane

I've downloaded it and tested. What it did was open a new cmd window rather than open the browser.

In a cmd window if I manually type 'start' followed by the URL for the IDE it does what you'd expect - opens a browser at the URL you fed to the 'start' cmd.

Screenshot shows what the behaviour currently is

[cid:e71f09be-e1ec-431c-9e8f-9c9acdadfc41]

Sam

________________________________

**Comment by SamFoster** [comment]
anavarre

just sent this to Dane - copying to you for info

--- snip ---

Dane

I've downloaded it and tested. What it did was open a new cmd window rather than open the browser.

In a cmd window if I manually type 'start' followed by the URL for the IDE it does what you'd expect - opens a browser at the URL you fed to the 'start' cmd.

Screenshot shows what the behaviour currently is

[cid:91eb397b-82c2-4dbb-8285-eaba9305a6ec]

Sam


________________________________

**Comment by danepowell** [comment]
Windows 🤦 

Okay, Powershell should be fixed now. I also confirmed that this works in other OSes. I don't know exactly what the problem was, but I speculate it's due to `start` being a shell built-in (or whatever is the Windows equivalent) rather than a true command, so it doesn't like taking arguments.

**Comment by danepowell** [comment]
After reviewing the [Symfony Process docs](https://symfony.com/doc/current/components/process.html), I'm pretty convinced there's actually a Symfony bug here. `new Process(['start', 'http://example'])` should start a browser, not a new shell, indicating that Symfony is dropping the argument somehow. But I'm not sure it's worth our time to chase that down.

**Pull request merged**